src/main/java/org/aksw/simba/squirrel/metadata/MetaDataHandler.java

```diff
@@ -35,18 +47,30 @@ public MetaDataHandler(String updateDatasetURI, String queryDatasetURI) {
+        Model model  = ModelFactory.createDefaultModel();
+        CrawleableUri uri = crawlingActivity.getCrawleableUri();
+
+        Node nodeResultGraph = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Result_"+ crawlingActivity.getGraphId(uri))));
+        Node nodeCrawlingActivity = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl(crawlingActivity.getId())));
+        Node Association = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Worker-checking-Crawl-guide")));
+        Node crawling = NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Crawl-guide")));
+
+        lstTriples.add(new Triple(nodeCrawlingActivity,RDF.type.asNode(),Prov.Activity.asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Prov.startedAtTime.asNode(), model.createTypedLiteral(crawlingActivity.getLocalDateTime(),XSDDatatype.XSDdateTime).asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Prov.endedAtTime.asNode(), model.createTypedLiteral(crawlingActivity.getLocalDateTime(),XSDDatatype.XSDdateTime).asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Sq.status.asNode(), NodeFactory.createLiteral(crawlingActivity.getState().toString())));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Sq.numberOfTriples.asNode(), model.createTypedLiteral(crawlingActivity.getNumTriples(),XSDDatatype.XSDint).asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Prov.wasAssociatedWith.asNode(), NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Worker_"+ String.valueOf(crawlingActivity.getWorker().getId()))))));
+        lstTriples.add(new Triple(nodeResultGraph, Prov.wasGeneratedBy.asNode(),nodeCrawlingActivity ));
+        lstTriples.add(new Triple(nodeResultGraph, RDFS.comment.asNode(),NodeFactory.createLiteral("GraphID where the content is stored")));
+        lstTriples.add(new Triple(nodeCrawlingActivity, Sq.hostedOn.asNode(),model.createResource(crawlingActivity.getHost()).asNode()));
+        lstTriples.add(new Triple(nodeCrawlingActivity,Prov.qualifiedAssociation.asNode(),Association));
+        lstTriples.add(new Triple(Association,RDF.type.asNode(),NodeFactory.createURI(Prov.Association.toString())));
+        lstTriples.add(new Triple(Association,Prov.agent.asNode(),NodeFactory.createURI(String.valueOf(crawlingActivity.getUrl("Worker_"+ String.valueOf(crawlingActivity.getWorker().getId()))))));
+        lstTriples.add(new Triple(Association,Prov.hadPlan.asNode(),crawling));
+        lstTriples.add(new Triple(crawling,RDF.type.asNode(),Prov.Plan.asNode()));
+        lstTriples.add(new Triple(crawling,RDF.type.asNode(),Prov.Entity.asNode()));
+        lstTriples.add(new Triple(crawling,Sq.steps.asNode(),NodeFactory.createLiteral(crawlingActivity.getHadPlan())));
 
         sink.openSinkForUri(dummyUri);
```

Comment:
This is not correct. I just saw that the Meta Data Handler is even creating its own sink assuming that it will be a SPARQL endpoint. As I already argued several times, we may have different sinks. So this meta data handler should use the that the worker has and it should use the [`sink.addMetaData`](https://github.com/varunmaitreya/Squirrel/blob/master/src/main/java/org/aksw/simba/squirrel/sink/Sink.java#L23) method which is already provided.